argparse_prompt.py

```diff
@@ -36,7 +37,10 @@ def add_argument(self, *args, prompt=True, secure=False, **kwargs):
                 del kwargs["default"]
 
             # Delegate to the parent class. Default must be '' in order to get the type function to be called
-            action = super().add_argument(*args, type=type, default="", **kwargs)
+            if(choices):
```

Comment:
I'd rather you extract the kwargs into a dictionary here so that we're not duplicating code.